### PR TITLE
Snocker via snorkel... take 2

### PR DESCRIPTION
**Pull request by DrPinkACN**
Some simple directions and a docker file to create a snorkel installed docker image that can be used to run snorkel installed containers. 

This image is also on my docker hub, and can be pulled with the following command:

```
docker pull drpinkacn/snocker:0.6.2
```

### CODE REVIEW

**Comment by DrPinkACN** [comment]
@ajratner I'm not sure what setting I am missing to add you as a reviewer. If you have any ideas, let me know.
image
:-/

**Comment by ajratner** [comment]
Hey sorry, didn't mean to make you jump through hoops!  No worries about assigning the reviewer- I'll just look at this PR as soon as I can this week

**Comment by vochicong** [comment]
Is this related to #323?

File: snocker/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+$ docker run -it --name snocker -p 8887:8887 -v ~/some/local/dir/mapped/to/snokel/projects:/home/snorkel/projects snocker:0.6.2
+## hit esc key sequence: ctrl+p+q
+```
+
+You can now execute a bash command in the running container, move the snorkel directory, and run snorkel.
+
+```
+$ docker exec -it snocker bash
+root@ab1234:/# cd /home/snorkel
+root@ab1234:/snorkel# ./run.sh
```
Comment (vochicong) [inline snocker/README.md]: It seems that `./run.sh` should be set as ENTRYPOINT in Dockerfile.

File: snocker/app/dockerfile [hunk 2]
```diff
@@ -0,0 +1,101 @@
+############################################################
```
Comment (vochicong) [inline snocker/app/dockerfile]: By convention, the file name should be Dockerfile (capital D).

File: snocker/app/run.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    read -p "CoreNLP [default] parser not found- install now?   " yn
+    case $yn in
+        [Yy]* ) echo "Installing parser..."; ./install-parser.sh;;
+        [Nn]* ) ;;
+    esac
+fi
+
+# Launch jupyter notebook!
+echo "Launching Jupyter Notebook..."
+jupyter notebook --port=8887 --ip=0.0.0.0 --allow-root
```
Comment (vochicong) [inline snocker/app/run.sh]: port 8888 is more usual?

**Comment by vochicong** [comment]
In order to try Snorkel, I tried this PR and have some small comments at https://github.com/HazyResearch/snorkel/pull/779/files.
Overall, I think that the folder structure is strange. 
For example, why don't you put the Dockerfile in the top folder?

File: snocker/README.md (hunk 1 above)
Comment (DrPinkACN) [reply snocker/README.md]: I have no problem with this change. It does assume that a user will know what the heck is going on ones the run dialog is asking to install coreNLP and returns a token. without explicitly initiating the process, one wont have a chance to explore the directory structure, or make the decision to launch.

File: snocker/app/dockerfile (hunk 2 above)
Comment (DrPinkACN) [reply snocker/app/dockerfile]: I think it works either way. I am not wed to any particular convention. Feel free to change/

File: snocker/app/run.sh (hunk 3 above)
Comment (DrPinkACN) [reply snocker/app/run.sh]: true, but it is all arbitrary. I was assuming that someone might have jupyter running locally. If so, 8888 could be spoken for. By choosing something less conventional, one needs simply to look at the URL and know that they are interacting with the containerized snorkel.

**Comment by ajratner** [comment]
@DrPinkACN @vochicong First of all, sorry for my delay in getting to these PRs, and thank you both so much for the contribs!!

After having looked at each other's PRs (this one and #795), any consensus on how to proceed?  I'm less familiar with Docker so discount my own opinion a bit... thoughts?

**Comment by DrPinkACN** [comment]
@ajratner I am happy to make the recommended changes to the structure of this PR made by @vochicong or combine ideas here with #795 . I am not a docker aficionado either, my build was informed by things I knew I needed, or wanted to avoid. the primary merit with the docker image this docker file builds is that phusion more of a fully functional linux kernel and using mini-conda python 2 build makes installing numba a little more headache free. These were not addressed in the comments above, so as long as some piece of those contributions survive, I'll feel like I helped 😄 .

All of this could be for not if we just maintain a working snorkel-docker image at docker hub. Thoughts @vochicong? I'm happy to collaborate on making the best dockerfile/image we can for this purpose.

**Comment by vochicong** [comment]
Thanks @DrPinkACN @ajratner for your input 👍 

I have a lot of work experiences with Docker and can help you to confidently build a good, clean Docker. But I'm not familiar with Snorkel and would like to learn more about it.

Basically, #795 can be thought as a refactoring of #779, for a cleaner Dockerfile, more standard structure, and easier usage. Can you @DrPinkACN give it a try?

I can also help to set up Travis CI so that it will automatically build and push up-to-date images to Docker Hub.

**Comment by bhancock8** [comment]
We dropped the ball on this one; not sure why it wasn't merged in before. Looks like it's been long enough now that the files may need a few small updates to pass the tests again. If anyone picks this up and updates it, happy to take another look and merge it in.

**Comment by vochicong** [comment]
I've reopened #795, which can be seen as a refactoring of this #779.

**Comment by vincentschen** [comment]
Closing—seems like this was updated in https://github.com/HazyResearch/snorkel/pull/795 !

**Pull request closed without merging**